File: backend-mcp/src/test/resources/mini-repos/java/src/main/java/com/example/DemoService.java
package com.example;

import com.example.repository.UserRepository;

/** Demo service doc. */
public class DemoService extends BaseService implements Runner, Runnable {

  private final UserRepository repository = new UserRepository();

  @Override
  public void run() {
    process("run", 1);
  }

  /** Process request with name and count. */
  public String process(String name, int count) {
    repository.findUser(name);
    helper(count);
    return name + count;
  }

  /** Overloaded process with default count. */
  public String process(String name) {
    return process(name, 1);
  }

  private void helper(int count) {
    this.log(count);
  }

  @Override
  protected void log(int value) {
    System.out.println("value=" + value);
  }
}

interface Runner {
  void run();
}

class BaseService {
  protected void baseMethod() {}

  protected void log(int value) {
    System.out.println("base=" + value);
  }
}

File: backend-mcp/src/test/resources/mini-repos/kotlin/src/main/kotlin/com/example/DemoService.kt
package com.example

/**
 * Kotlin demo service with inheritance and overloaded methods.
 */
class DemoService(
    private val repository: UserRepository = UserRepository()
) : BaseService(), Runner {

  override fun run() {
    process("kotlin", 2)
  }

  override fun process(name: String, count: Int): String {
    log(count)
    repository.save(name)
    return "$name-$count"
  }

  fun processOnce(name: String): String = process(name, 1)

  override fun log(value: Int) {
    println("value=$value")
  }
}
